File: server/internal/schema/note.go
package schema

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Note struct {
	ID               primitive.ObjectID `bson:"_id,omitempty" json:"id,omitempty"`
	JobApplicationID primitive.ObjectID `bson:"jobApplicationID" json:"jobApplicationID" binding:"required"`
	Content          string             `bson:"name" json:"name" binding:"required"`
	Timestamp        time.Time          `bson:"timestamp" json:"timestamp" binding:"required"`
}

File: client/src/lib/types/note.ts
/**
 * Note-related types matching backend schema
 */

export interface Note {
  id?: string;
  jobApplicationID: string;
  content: string;
  timestamp: string;
}

export interface CreateNoteRequest {
  jobApplicationID: string;
  content: string;
  timestamp?: string;
}

export interface UpdateNoteRequest {
  id?: string;
  jobApplicationID?: string;
  content?: string;
  timestamp?: string;
}

export interface NoteFilters {
  jobApplicationID?: string;
}